tap_messagebird/streams.py

```diff
@@ -70,11 +72,52 @@ def get_child_context(
             "_sdc_conversations_id": record["id"],
         }
 
+    def get_records(self, context: dict | None) -> Iterable[dict[str, Any]]:
+        """Return a generator of record-type dictionary objects.
+
+        Each record emitted should be a dictionary of property names to their values.
+
+        Args:
+            context: Stream partition or context dictionary.
+
+        Yields:
+            One item per (possibly processed) record in the API.
+        """
+        starting_replication_key_value = self.get_starting_timestamp(context)
+        if starting_replication_key_value is None:
+            # Shouldn't be possible, but mypy complains
+            raise ValueError(  # noqa: TRY003
+                "No starting replication key value found.",
+            )
+        starting_replication_key_value = pendulum.instance(
+            starting_replication_key_value,
+        )
+
+        for record in self.request_records(context):
+            transformed_record = self.post_process(record, context)
+            if transformed_record is None:
+                # Record filtered out during post_process()
+                continue
+            record_last_received_datetime: pendulum.DateTime = cast(
+                pendulum.DateTime,
+                pendulum.parse(record[self.replication_key]),
+            )
+            # Conversations are returned in descending order, so we can stop
+            # There's no filtering paramater just this default ordering
+            if record_last_received_datetime < starting_replication_key_value:
+                self.logger.info(
+                    f"Hit a record with a {str(record_last_received_datetime)=}"
+                    f"< {str(starting_replication_key_value)=}. Stopping.",
+                )
```

Comment:
```suggestion
                self.logger.info(
                    "Breaking after hitting a record with replication key %s < %s",
                    record_last_received_datetime,
                    starting_replication_key_value,
                )
```

Reply:
It's not hate 😅, just convention and improves performance without a large dev cost when filtering out logs by level